### PR TITLE
Updated Summary plot defaults and SDarn 

### PR DESCRIPTION
**Pull request by mts299**
I wanted this to be two PR's but due to a time crunch and I needed to produce summary plots for researchers, I did all the changes in one this PR. 

This PR addresses the following:
- [x] updated DarnRead, DarnWrite to SDarnRead, SDarnWrite
- [x] Reading methods returns a `dict` and not a `dmap` structure
- [x] Writing methods take either a `dmap` structure or `dict`
- [x] Updated Summary plot defaults (see list below)
- [x] fixed a couple bugs/typos
- [x] Updated Borealis conversion code to accept the new IO changes 

Test example of how to call the new method names and summary plot defaults:
``` python
import pydarn
import matplotlib.pyplot as plt 

fitacf_file = "20180101.C0.cly.fitacf"
darn_read = pydarn.SDarnRead(fitacf_file)
fitacf_data = darn_read.read_fitacf()
pydarn.RTP.plot_summary(fitacf_data, beam_num=3)
#plt.show()
plt.savefig('default_summary_plot.png')
``` 

default_summary_plot

And from the pyDARN developers meeting the new defaults were also updated in this PR. 
The updates are as follows:
- [x] spectral width default range 0-250
- [x] elevation default range 0-45 
- [x] default color map on SNR to plasma 
- [x] default colormap for velocity with red and blue with a non-white/grey middle color 
- [x] Removed Search Noise 
- [x] updated other color maps to be more readable but colorblind-friendly  

@mardet987 you may need to change some of the Borealis conversion methods based on these new changes. You can do that in this branch. I can also help as long as you can unit test it.

### CODE REVIEW

**Comment by mardet987** [comment]
I updated the Borealis code for new SDarn class names and completed unit and integration tests.

**Comment by billetd** [comment]
Looks all good to me. Everything worked just fine as before.

File: pydarn/plotting/color_maps.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import matplotlib.colors as mcol
+
+class PyDARNColormaps():
+    PYDARN_VELOCITY = mcol.LinearSegmentedColormap.from_list("pydarn_velocity",["darkred", "r",'pink',"b", "darkblue"])
+
+    PYDARN_VIRIDIS = mcol.LinearSegmentedColormap.from_list("pydarn_virdis",["indigo", "midnightblue","navy", "mediumblue",
```
[the remark this answers is not in the document]
Comment (mts299) [reply pydarn/plotting/color_maps.py]: fixed!

**Comment by mts299** [comment]
Fixed up the docstrings on the plots, changed more defaults upon @ecbland and Kathryn's request. 
Here are the new default summary plots:
Figure_1

Changes are as follows:
- [x] color maps 
- [x] removed search noise
- [x] change ranges for parameters (SNR: 0-45, Vel: 200- -200, Spectral Wdith: 0-250, Elevation: 0-45)
- [x] truncate range gates to 0-75  (Kathryn's request)
- [x] watermark for not publication use (Kathryn's request)
- [x] cleaned up color bars to look nicer and time-series y-axis

I did change the default colormap for Range-time plots to pyDARN's own color map:
Figure_2

**Comment by mardet987** [comment]
I think this all looks great! Are you still waiting on another reviewer @mts299?

**Comment by mts299** [comment]
@mardet987 Sorry for got to reply. 

Yes, I think it is good to merge if @billetd and you are good with the testing/review.

**Comment by mts299** [comment]
Since this PR is still open I also added `xcfd` as an optional field for RAWACF format. 
Sorry for the typo in the commit message.

**Pull request merged**